File: backend/QuizHubApplication/DTOs/Responses/QuestionDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Responses
{
    public record QuestionDto(int Id, string Text, int Points, string QuestionType, string? CorrectFillInAnswer,
        List<AnswerOptionDto> AnswerOptions)
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/quiz/models/QuestionDto.ts
import { AnswerOptionDto } from './AnswerOptionDto';

export interface QuestionDto {
    id: number;
    text: string;
    points: number;
    questionType: string;
    correctFillInAnswer?: string; // ? jer može biti null
    answerOptions: AnswerOptionDto[];
}
